src/Xamarin.Android.Build.Tasks/Tasks/GetAdditionalResourcesFromAssemblies.cs

```diff
@@ -172,7 +173,7 @@ bool ExtractArchive (string url, string file, string contentDir)
 					}
 				}
 				catch (Exception e) {
-					LogCodedError ("XA5209", "Unzipping failed. Please download {0} and extract it to the {1} directory.", url, contentDir);
+					LogCodedError ("XA5209", "Unzipping failed while attempting to extract {0} into {1}{2}", file, contentDir, Path.DirectorySeparatorChar);
```

Comment:
What is the purpose of an error message? It's not (necessarily) *just* to say "something went wrong." If that were the sole purpose, we could print out the `Reason` message and be done with it.

The purpose of the error message is *also* to inform the user of *what* the problem is in such a way that *they can fix it*.

Your updated message doesn't satisfy that purpose. We just know that unzipping *some file* failed for *some reason*. We don't know what to do to fix it. Should we clean and rebuild the project? Sacrifice a chicken? Burn the world? Who knows! The error message certainly doesn't say....

As such, the previous message is much better.